huaweicloud/services/dcs/data_source_huaweicloud_dcs_flavors.go

```diff
@@ -0,0 +1,166 @@
+package dcs
+
+import (
+	"github.com/chnsz/golangsdk/openstack/dcs/v2/flavors"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+// DataSourceDcsFlavorsV2 the function is used for data source 'huaweicloud_dcs_flavors'.
+func DataSourceDcsFlavorsV2() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDcsFlavorsV2Read,
```

Comment:
please use **ReadContext**

Reply:
If the context is not used in the data source, use `Read` is simpler.

Reply:
you can get the reason form [v2 Upgrade Guide](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#more-support-for-context-context).

Reply:
nice